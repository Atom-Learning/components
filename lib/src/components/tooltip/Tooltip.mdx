---
title: Tooltip
component: Tooltip,Tooltip.Trigger,Tooltip.Content,Tooltip.Provider, Tooltip.Portal
description: A component that displays information related to an element when the element receives keyboard focus or the mouse hovers over it
category: Surfaces
---

`Tooltip` exports 4 components that combine to create our tooltip. The `Tooltip.Trigger` requires a React component child (or passed to `as`) to allow the necessary event bindings and accessible properties to be cloned.

Also, note that the component must accept a `ref`.

`Tooltip.Provider` is **required to avoid regressions** and should be present at the top level of the app. 

Read more about the underlying UI component on the [Radix UI documentation site](https://radix-ui.com/primitives/docs/components/tooltip).

```tsx preview
<Tooltip.Provider>
  <Tooltip>
    <Tooltip.Trigger>
      <Text>Hover on me</Text>
    </Tooltip.Trigger>
    <Tooltip.Content>This is the tooltip content, hello</Tooltip.Content>
  </Tooltip>
</Tooltip.Provider>
```
