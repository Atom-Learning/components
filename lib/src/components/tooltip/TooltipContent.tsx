import { Arrow, Content } from '@radix-ui/react-tooltip'
import * as React from 'react'

import { styled } from '~/stitches'
import {
  slideDownAndFade,
  slideLeftAndFade,
  slideRightAndFade,
  slideUpAndFade
} from '~/utilities'

const StyledContent = styled(Content, {
  backgroundColor: '$tonal500',
  borderRadius: '$0',
  boxShadow: '$0',
  color: 'white',
  fontFamily: '$body',
  fontSize: '$sm',
  lineHeight: 1.5,
  px: '$3',
  py: '$2',
  '@allowMotion': {
    animationDuration: '75ms',
    animationTimingFunction: 'cubic-bezier(0.16, 1, 0.3, 1)',
    willChange: 'transform, opacity',
    '&[data-state="delayed-open"]': {
      '&[data-side="top"]': { animationName: slideDownAndFade },
      '&[data-side="right"]': { animationName: slideLeftAndFade },
      '&[data-side="bottom"]': { animationName: slideUpAndFade },
      '&[data-side="left"]': { animationName: slideRightAndFade }
    }
  },
  variants: {
    size: {
      sm: { maxWidth: '100px' },
      md: { maxWidth: '250px' },
      lg: { maxWidth: '400px' }
    }
  }
})

const StyledArrow = styled(Arrow, {
  fill: '$tonal500',
  '[data-align="end"] &': { mr: '$2' },
  '[data-align="start"] &': { ml: '$2' }
})

type TooltipContentProps = React.ComponentProps<typeof StyledContent> &
  React.ComponentProps<typeof Content>

export const TooltipContent: React.FC<TooltipContentProps> = ({
  children,
  side = 'top',
  sideOffset = 4,
  size = 'md',
  ...remainingProps
}) => (
  <StyledContent
    side={side}
    sideOffset={sideOffset}
    size={size}
    {...remainingProps}
  >
    {children}
    <StyledArrow />
  </StyledContent>
)
