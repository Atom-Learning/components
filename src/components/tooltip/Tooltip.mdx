---
title: Tooltip
component: Tooltip,Tooltip.Trigger,Tooltip.Content
description: A popup that displays information related to an element when the element receives keyboard focus or the mouse hovers over it.
category: Surfaces
---

`Tooltip` exports 3 components that combine to create our tooltip. The `Tooltip.Trigger` requires a React component child (or passed to `as`) to allow the necessary event bindings and accessible properties to be cloned, also note that the component must accept a `ref`.

Read more about the underlying UI component on the [Radix UI documentation site](https://radix-ui.com/primitives/docs/components/tooltip).

```jsx preview
<Tooltip>
  <Tooltip.Trigger>
    <Text>Hover on me</Text>
  </Tooltip.Trigger>
  <Tooltip.Content>This is the tooltip content, hello</Tooltip.Content>
</Tooltip>
```
