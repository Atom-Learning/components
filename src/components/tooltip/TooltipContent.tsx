import { Arrow, Content } from '@radix-ui/react-tooltip'
import * as React from 'react'

import { keyframes, styled } from '~/stitches'

const scaleIn = keyframes({
  '0%': { opacity: 0, transform: 'scale(0.8)' },
  '100%': { opacity: 1, transform: 'scale(1)' }
})

const StyledContent = styled(Content, {
  animation: `${scaleIn} 75ms ease-out`,
  backgroundColor: '$tonal600',
  borderRadius: '$0',
  color: 'white',
  fontFamily: '$sans',
  fontSize: '$sm',
  px: '$2',
  py: '$1',
  transformOrigin: 'var(--radix-tooltip-content-transform-origin)'
})

const StyledArrow = styled(Arrow, {
  fill: '$tonal600'
})

type TooltipContentProps = React.ComponentProps<typeof Content>

export const TooltipContent: React.FC<TooltipContentProps> = ({
  sideOffset = 8,
  side = 'top',
  children,
  ...props
}) => (
  <StyledContent side={side} sideOffset={sideOffset} {...props}>
    {children}
    <StyledArrow />
  </StyledContent>
)
