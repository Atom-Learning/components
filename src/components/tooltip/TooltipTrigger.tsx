import { Slot } from '@radix-ui/react-slot'
import { Trigger } from '@radix-ui/react-tooltip'
import * as React from 'react'

type TooltipTriggerProps = React.ComponentProps<typeof Trigger>

export const TooltipTrigger: React.FC<TooltipTriggerProps> = ({
  children,
  ...remainingProps
}) => (
  <Trigger as={Slot} {...remainingProps}>
    {children}
  </Trigger>
)
