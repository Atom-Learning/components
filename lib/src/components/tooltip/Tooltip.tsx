import { Portal, Provider, Root, Trigger } from '@radix-ui/react-tooltip'
import * as React from 'react'

import { styled } from '~/stitches'

import { TooltipContent } from './TooltipContent'

type TooltipProps = React.ComponentProps<typeof Root>

export const Tooltip: React.FC<TooltipProps> & {
  Content: typeof TooltipContent
  Portal: typeof Portal
  Trigger: typeof Trigger
  Provider: typeof Provider
} = ({ children, delayDuration = 350, ...remainingProps }) => (
  <Root delayDuration={delayDuration} {...remainingProps}>
    {children}
  </Root>
)

Tooltip.Content = TooltipContent
Tooltip.Trigger = styled(Trigger, {})
Tooltip.Portal = Portal
Tooltip.Provider = Provider

Tooltip.displayName = 'Tooltip'
