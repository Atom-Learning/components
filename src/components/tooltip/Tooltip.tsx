import { Root } from '@radix-ui/react-tooltip'
import * as React from 'react'

import { TooltipContent } from './TooltipContent'
import { TooltipTrigger } from './TooltipTrigger'

type TooltipProps = React.ComponentProps<typeof Root>

export const Tooltip: React.FC<TooltipProps> & {
  Content: typeof TooltipContent
  Trigger: typeof TooltipTrigger
} = ({ children, delayDuration = 100, ...remainingProps }) => (
  <Root delayDuration={delayDuration} {...remainingProps}>
    {children}
  </Root>
)

Tooltip.Content = TooltipContent
Tooltip.Trigger = TooltipTrigger

Tooltip.displayName = 'Tooltip'
