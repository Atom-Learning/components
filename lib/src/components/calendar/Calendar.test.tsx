import { fireEvent, render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Tooltip } from '../tooltip'
import { Calendar } from '.'
// NOTE: We need to set the date or the 'isToday' class on individual
// dates will cause these snapshots to fail every new day this test is
// run. Setting this system time means this is kept consistent.
Date.now = jest.fn(() => Date.parse('2021-12-14'))

const props = {
  onDateSelected: jest.fn(),
  date: new Date('12/25/21'),
  setYear: jest.fn()
}

const Wrapper = ({ children }) => (
  <Tooltip.Provider>{children}</Tooltip.Provider>
)

describe('Calendar component', () => {
  it('renders', async () => {
    const { container } = render(
      <Wrapper>
        <Calendar {...props} />
      </Wrapper>
    )

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <Wrapper>
        <Calendar {...props} />
      </Wrapper>
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('changes month when change month arrows are pressed', async () => {
    render(
      <Wrapper>
        <Calendar {...props} />
      </Wrapper>
    )

    const arrow = await screen.findByLabelText('Next month')
    fireEvent.click(arrow)

    expect(await screen.queryByText('Dec 2021')).not.toBeInTheDocument()
  })

  it('opens the change year view when year is clicked', async () => {
    render(
      <Wrapper>
        <Calendar {...props} />
      </Wrapper>
    )

    const yearButton = await screen.findByText('Dec 2021')
    fireEvent.click(yearButton)

    expect(await screen.queryByText('Dec 2021')).not.toBeInTheDocument()
    expect(await screen.findByText('2020')).toBeVisible()
  })

  it('changes year when change year arrows are pressed', async () => {
    render(
      <Wrapper>
        <Calendar {...props} />
      </Wrapper>
    )

    const yearButton = await screen.findByText('Dec 2021')
    fireEvent.click(yearButton)

    const arrow = await screen.findByLabelText('Next year')
    fireEvent.click(arrow)

    expect(await screen.queryByText('2021')).not.toBeInTheDocument()
    expect(await screen.findByText('2025')).toBeVisible()
  })

  it('cannot select year later than max date year', async () => {
    render(
      <Wrapper>
        <Calendar {...props} maxDate={new Date('2025-01-01')} />
      </Wrapper>
    )

    const yearButton = await screen.findByText('Dec 2021')
    fireEvent.click(yearButton)

    const arrow = await screen.findByLabelText('Next year')
    fireEvent.click(arrow)

    expect(await screen.queryByText('2021')).not.toBeInTheDocument()
    expect(await screen.findByText('2025')).toBeVisible()
    expect(await screen.queryByText('2026')).not.toBeInTheDocument()
  })

  it('cannot select year earlier than min date year', async () => {
    render(
      <Wrapper>
        <Calendar {...props} minDate={new Date('2000-01-01')} />
      </Wrapper>
    )

    const yearButton = await screen.findByText('Dec 2021')
    fireEvent.click(yearButton)

    const arrow = await screen.findByLabelText('Previous year')
    fireEvent.click(arrow)

    expect(await screen.queryByText('2021')).not.toBeInTheDocument()
    expect(await screen.findByText('2000')).toBeVisible()
    expect(await screen.queryByText('1999')).not.toBeInTheDocument()
  })

  it('renders translated weekday names', async () => {
    render(
      <Wrapper>
        <Calendar
          {...props}
          weekdayNames={[
            'Måndag',
            'Tisdag',
            'Onsdag',
            'Torsdag',
            'Fredag',
            'Lördag',
            'Söndag'
          ]}
        />
      </Wrapper>
    )

    expect(await screen.findByText('Måndag')).toBeVisible()
  })

  it('renders translated month names', async () => {
    render(
      <Wrapper>
        <Calendar
          {...props}
          date={new Date('08/01/21')}
          monthNames={[
            'Januri',
            'Februari',
            'Mars',
            'April',
            'Maj',
            'Juni',
            'Juli',
            'Augusti',
            'September',
            'Oktober',
            'November',
            'December'
          ]}
        />
      </Wrapper>
    )

    expect(await screen.findByText('Augusti 2021')).toBeVisible()
  })
})
