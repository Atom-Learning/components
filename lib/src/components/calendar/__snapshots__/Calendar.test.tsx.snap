// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar component renders 1`] = `
@media  {
  .c-dhzjXW {
    display: flex;
  }

  .c-iPabyW {
    align-items: center;
    -webkit-appearance: none;
    appearance: none;
    background: white;
    border: unset;
    border-radius: var(--radii-0);
    box-sizing: border-box;
    cursor: pointer;
    display: flex;
    justify-content: center;
    padding: unset;
    transition: all 100ms ease-out;
  }

  .c-iPabyW[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-dbrbZt {
    display: inline-block;
    fill: none;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }

  .c-hVoQtf {
    background-color: var(--colors-tonal500);
    border-radius: var(--radii-0);
    box-shadow: var(--shadows-0);
    color: white;
    font-family: var(--fonts-body);
    font-size: var(--fontSizes-sm);
    line-height: 1.5;
    white-space: normal;
    padding-left: var(--space-3);
    padding-right: var(--space-3);
    padding-top: var(--space-2);
    padding-bottom: var(--space-2);
    z-index: 10;
  }

@media (prefers-reduced-motion: no-preference) {
    .c-hVoQtf {
      animation-duration: 75ms;
      animation-timing-function: cubic-bezier(0.16, 1, 0.3, 1);
      will-change: transform, opacity;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-hVoQtf[data-state="delayed-open"][data-side="top"] {
      animation-name: k-ftcNPK;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-hVoQtf[data-state="delayed-open"][data-side="right"] {
      animation-name: k-hbaHED;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-hVoQtf[data-state="delayed-open"][data-side="bottom"] {
      animation-name: k-daNevH;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-hVoQtf[data-state="delayed-open"][data-side="left"] {
      animation-name: k-fvyGIa;
    }
}

  .c-kaaUSm {
    align-items: center;
    background: unset;
    border: unset;
    border-radius: var(--radii-0);
    cursor: pointer;
    display: flex;
    font-family: var(--fonts-body);
    font-weight: 600;
    justify-content: center;
    padding: unset;
    text-decoration: none;
    transition: all 100ms ease-out;
    white-space: nowrap;
    width: max-content;
  }

  .c-kaaUSm[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-fPxPMR {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    grid-gap: var(--space-1) var(--space-2);
  }

  .c-dyvMgW {
    font-family: var(--fonts-body);
    font-weight: 400;
    margin: 0;
  }

  .c-dyvMgW > .c-dyvMgW:before,
  .c-dyvMgW > .c-dyvMgW:after {
    display: none;
  }

  .c-gNRHdp {
    background: transparent;
    border: none;
    border-radius: var(--radii-round);
    color: var(--colors-tonal900);
    cursor: pointer;
    font-family: var(--fonts-body);
    font-size: var(--fontSizes-sm);
    height: var(--sizes-3);
    width: var(--sizes-3);
    padding: 0;
    transition: all 75ms;
  }

  .c-gNRHdp:hover {
    background: var(--colors-tonal100);
  }

  .c-gNRHdp:active {
    color: white;
    background: var(--colors-primary);
  }

  .c-gNRHdp:focus {
    outline: 2px solid var(--colors-primary);
    outline-offset: 2px;
  }
}

@media  {
  .c-iPabyW-PrXKS-size-md {
    height: var(--sizes-4);
    width: var(--sizes-4);
  }

  .c-dbrbZt-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }

  .c-hVoQtf-gWQnqe-size-md {
    max-width: 250px;
  }

  .c-dhzjXW-iTKOFX-direction-column {
    flex-direction: column;
  }

  .c-dhzjXW-gkvzBZ-gap-4 {
    gap: var(--space-4);
  }

  .c-dhzjXW-jroWjL-align-center {
    align-items: center;
  }

  .c-kaaUSm-dXsdwv-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
    height: var(--sizes-4);
    padding-left: var(--space-5);
    padding-right: var(--space-5);
    gap: var(--space-3);
  }

  .c-kaaUSm-dXsdwv-size-md .c-dbrbZt {
    height: 20px;
    width: 20px;
  }

  .c-dyvMgW-bndJoy-size-sm {
    font-size: var(--fontSizes-sm);
    line-height: 1.53;
  }

  .c-dyvMgW-bndJoy-size-sm::before {
    content: '';
    margin-bottom: -0.4056em;
    display: table;
  }

  .c-dyvMgW-bndJoy-size-sm::after {
    content: '';
    margin-top: -0.4056em;
    display: table;
  }

  .c-gNRHdp-huXVXM-isOutsideMonth-true {
    color: var(--colors-tonal200);
  }

  .c-gNRHdp-hztIub-isToday-true {
    background: var(--colors-tonal100);
  }
}

@media  {
  .c-iPabyW-hXLsaq-cv {
    background: transparent;
    color: var(--colors-tonal300);
  }

  .c-iPabyW-hXLsaq-cv:not(:disabled):hover,
  .c-iPabyW-hXLsaq-cv:not(:disabled):focus {
    color: var(--colors-primaryMid);
  }

  .c-iPabyW-hXLsaq-cv:not(:disabled):active {
    color: var(--colors-primaryDark);
  }

  .c-kaaUSm-jlrmAu-cv {
    background: white;
    color: var(--colors-primary);
  }

  .c-kaaUSm-jlrmAu-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-jlrmAu-cv:not([disabled]):hover,
  .c-kaaUSm-jlrmAu-cv:not([disabled]):focus {
    background: rgba(255, 255, 255, 0.9);
    color: var(--colors-primary);
  }

  .c-kaaUSm-jlrmAu-cv:not([disabled]):active {
    background: rgba(255, 255, 255, 0.75);
  }
}

@media  {
  .c-PJLV-ijkIlaT-css {
    position: relative;
    width: min-content;
    margin-top: calc(var(--space-1)*-1);
  }

  .c-dhzjXW-ihOOvbW-css {
    position: absolute;
    top: 0;
    right: calc(var(--space-1)*-1);
  }

  .c-dhzjXW-idNqCDY-css {
    height: var(--sizes-4);
  }

  .c-kaaUSm-ikwYYXS-css {
    padding-left: 0;
    padding-right: 0;
    color: var(--colors-tonal600);
  }

  .c-dyvMgW-idQvJRf-css {
    font-weight: 600;
    text-align: center;
  }
}

<div>
  <div
    class="c-PJLV c-PJLV-ijkIlaT-css"
  >
    <div
      class="c-dhzjXW c-dhzjXW-ihOOvbW-css"
    >
      <button
        aria-label="Previous month"
        class="c-iPabyW c-iPabyW-PrXKS-size-md c-iPabyW-hXLsaq-cv c-PJLV"
        data-state="closed"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="c-dbrbZt c-dbrbZt-dMrjnF-size-md c-dbrbZt-iPJLV-css"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="14 18 8 12 14 6 14 6"
          />
        </svg>
      </button>
      <button
        aria-label="Next month"
        class="c-iPabyW c-iPabyW-PrXKS-size-md c-iPabyW-hXLsaq-cv c-PJLV"
        data-state="closed"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="c-dbrbZt c-dbrbZt-dMrjnF-size-md c-dbrbZt-iPJLV-css"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="10 6 16 12 10 18 10 18"
          />
        </svg>
      </button>
    </div>
    <div
      class="c-dhzjXW c-dhzjXW-iTKOFX-direction-column c-dhzjXW-gkvzBZ-gap-4"
    >
      <div
        class="c-dhzjXW c-dhzjXW-jroWjL-align-center c-dhzjXW-idNqCDY-css"
      >
        <button
          class="c-kaaUSm c-kaaUSm-dXsdwv-size-md c-kaaUSm-jlrmAu-cv c-kaaUSm-ikwYYXS-css"
          type="button"
        >
          Dec
           
          2021
        </button>
      </div>
      <div
        class="c-fPxPMR"
      >
        <span
          class="c-dyvMgW c-dyvMgW-bndJoy-size-sm c-dyvMgW-idQvJRf-css"
        >
          Sun
        </span>
        <span
          class="c-dyvMgW c-dyvMgW-bndJoy-size-sm c-dyvMgW-idQvJRf-css"
        >
          Mon
        </span>
        <span
          class="c-dyvMgW c-dyvMgW-bndJoy-size-sm c-dyvMgW-idQvJRf-css"
        >
          Tue
        </span>
        <span
          class="c-dyvMgW c-dyvMgW-bndJoy-size-sm c-dyvMgW-idQvJRf-css"
        >
          Wed
        </span>
        <span
          class="c-dyvMgW c-dyvMgW-bndJoy-size-sm c-dyvMgW-idQvJRf-css"
        >
          Thu
        </span>
        <span
          class="c-dyvMgW c-dyvMgW-bndJoy-size-sm c-dyvMgW-idQvJRf-css"
        >
          Fri
        </span>
        <span
          class="c-dyvMgW c-dyvMgW-bndJoy-size-sm c-dyvMgW-idQvJRf-css"
        >
          Sat
        </span>
      </div>
      <div
        class="c-fPxPMR"
      >
        <button
          aria-label="Sun Nov 28 2021"
          aria-pressed="false"
          class="c-gNRHdp c-gNRHdp-huXVXM-isOutsideMonth-true"
          role="button"
          type="button"
        >
          28
        </button>
        <button
          aria-label="Mon Nov 29 2021"
          aria-pressed="false"
          class="c-gNRHdp c-gNRHdp-huXVXM-isOutsideMonth-true"
          role="button"
          type="button"
        >
          29
        </button>
        <button
          aria-label="Tue Nov 30 2021"
          aria-pressed="false"
          class="c-gNRHdp c-gNRHdp-huXVXM-isOutsideMonth-true"
          role="button"
          type="button"
        >
          30
        </button>
        <button
          aria-label="Wed Dec 01 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          1
        </button>
        <button
          aria-label="Thu Dec 02 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          2
        </button>
        <button
          aria-label="Fri Dec 03 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          3
        </button>
        <button
          aria-label="Sat Dec 04 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          4
        </button>
        <button
          aria-label="Sun Dec 05 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          5
        </button>
        <button
          aria-label="Mon Dec 06 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          6
        </button>
        <button
          aria-label="Tue Dec 07 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          7
        </button>
        <button
          aria-label="Wed Dec 08 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          8
        </button>
        <button
          aria-label="Thu Dec 09 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          9
        </button>
        <button
          aria-label="Fri Dec 10 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          10
        </button>
        <button
          aria-label="Sat Dec 11 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          11
        </button>
        <button
          aria-label="Sun Dec 12 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          12
        </button>
        <button
          aria-label="Mon Dec 13 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          13
        </button>
        <button
          aria-label="Tue Dec 14 2021"
          aria-pressed="false"
          class="c-gNRHdp c-gNRHdp-hztIub-isToday-true"
          role="button"
          type="button"
        >
          14
        </button>
        <button
          aria-label="Wed Dec 15 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          15
        </button>
        <button
          aria-label="Thu Dec 16 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          16
        </button>
        <button
          aria-label="Fri Dec 17 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          17
        </button>
        <button
          aria-label="Sat Dec 18 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          18
        </button>
        <button
          aria-label="Sun Dec 19 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          19
        </button>
        <button
          aria-label="Mon Dec 20 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          20
        </button>
        <button
          aria-label="Tue Dec 21 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          21
        </button>
        <button
          aria-label="Wed Dec 22 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          22
        </button>
        <button
          aria-label="Thu Dec 23 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          23
        </button>
        <button
          aria-label="Fri Dec 24 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          24
        </button>
        <button
          aria-label="Sat Dec 25 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          25
        </button>
        <button
          aria-label="Sun Dec 26 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          26
        </button>
        <button
          aria-label="Mon Dec 27 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          27
        </button>
        <button
          aria-label="Tue Dec 28 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          28
        </button>
        <button
          aria-label="Wed Dec 29 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          29
        </button>
        <button
          aria-label="Thu Dec 30 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          30
        </button>
        <button
          aria-label="Fri Dec 31 2021"
          aria-pressed="false"
          class="c-gNRHdp"
          role="button"
          type="button"
        >
          31
        </button>
        <button
          aria-label="Sat Jan 01 2022"
          aria-pressed="false"
          class="c-gNRHdp c-gNRHdp-huXVXM-isOutsideMonth-true"
          role="button"
          type="button"
        >
          1
        </button>
      </div>
    </div>
  </div>
</div>
`;
